### DISCUSSION TITLE
VSCode token expired, can't renew

### DISCUSSION THREAD

**Discussion by linusnorton** [community/community#43183] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

When I start up VSCode I tells me the token has expired:

GitHub Copilot could not connect to server. Extension activation failed: "Your GitHub token is invalid. Please sign out from your GitHub account using VSCode UI and try again."

But there is no way to bring back the VSCode UI to re-auth. I've tried deleting the extension and re-installing but it has the same issue. Where is the token stored? Maybe I can just delete it from disk?

**Comment by LuisContrerasGlz** [comment]
you can try to sign out from your GitHub account in Visual Studio Code and then sign back in or ry deleting the extension and then re-installing it :-) 

Hope this helps

**Reply by linusnorton** [reply]
How do I get the VSCode browser up to do that?

**Reply by LuisContrerasGlz** [reply]
Maybe it could be easier to uninstall it and then configure it again? it is what it occurs to me

**Reply by linusnorton** [reply]
I have tried but after re-installing it doesn't give me the option to do the configuration again it just seems to load my old configuration. I've tried deleting the whole copilot folder in ~/.vscode/extensions but it didn't seem to help. I'm not sure where it's storing the token

**Reply by LuisContrerasGlz** [reply]
Try deleting the Visual Studio Code configuration files from your system. These files are typically stored in the ~/.config/Code folder on Linux, the ~/Library/Application Support/Code folder on macOS, and the %APPDATA%\Code folder on Windows.

**Reply by linusnorton** [reply]
I'd rather not nuke my entire VSCode config. I'll see if I can find the token in that folder

**Reply by LuisContrerasGlz** [reply]
sounds good, hope you can fix it :-) best regards

**Comment by iamcjwilliams** [comment]
@linusnorton Please visit here to trouble your copilot issue: https://docs.github.com/en/copilot/troubleshooting-github-copilot

**Reply by linusnorton** [reply]
Not much of use in there. The logs are:

[INFO] [auth] [2023-01-03T12:46:30.770Z] Failed to get copilot token due to 401 status
[ERROR] [default] [2023-01-03T12:46:30.772Z] GitHub Copilot could not connect to server. Extension activation failed: "Your GitHub token is invalid. Please sign out from your GitHub account using VSCode UI and try again."

**Comment by silostack** [comment]
getting this as well

**Reply by linusnorton** [reply]
I haven't found a solution unfortunately

**Comment by kimadeline** [comment, marked as answer]
Hi @linusnorton and @silostack, can you try logging out from the "Accounts" view in VS Code:

![image](https://user-images.githubusercontent.com/51720070/210896921-b9abf1f0-3856-4cfa-bf6a-24041b64234c.png)

If the account icon doesn't appear, right-click on the activity bar and select "Accounts":

![image](https://user-images.githubusercontent.com/51720070/210897015-1b82d15d-e6c0-449a-bf02-a967e8fc7b0a.png)

**Reply by silostack** [reply]
yea that didn't work

**Reply by linusnorton** [reply]
I'd hidden the Activity bar. I got it working by clicking on the Person icon, going to log out, restarting VSCode, then logging in again when prompted by the VSCode extension.

Thanks

**Reply by JasonToups** [reply]
Thanks @linusnorton , that worked for me 👍

**Reply by krunalpandav** [reply]
Thanks @linusnorton ,It's worked

**Comment by silostack** [comment]
Got in touch w/support. I had to fix the Copilot settings here: https://github.com/settings/copilot. Super weird cause I'm pretty sure I had check that yesterday but... 🤷‍♂️ Anyway, working now!

**Comment by christopher-danz** [comment]
I had the same issue. 
What worked for me was to click on the person-icon that is shown above and sign out of my GitHub account. Then close VSCode and open it again. When I clicked on the person-icon again, I got the option to connect with a GitHub account. I clicked on it and then a pop-up window showed up, asking me if I want to connect my device to GitHub. When allowing, I got redirected into a browser tab where a 6-figure code had to paste in. Simultaneously, VSCode showed another pop-up window with this code that I pasted in and then it worked.
